x-pack/plugins/ml/server/models/data_frame_analytics/index_patterns.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClientContract } from 'kibana/server';
+import { IIndexPattern } from 'src/plugins/data/server';
+
+export class IndexPatternHandler {
+  constructor(private savedObjectsClient: SavedObjectsClientContract) {}
+  // returns a id based on an index pattern name
+  async getIndexPatternId(indexName: string) {
+    try {
+      const response = await this.savedObjectsClient.find<IIndexPattern>({
+        type: 'index-pattern',
+        perPage: 10,
+        search: `"${indexName}"`,
+        searchFields: ['title'],
+        fields: ['title'],
+      });
+
+      const ip = response.saved_objects.find(
+        obj => obj.attributes.title.toLowerCase() === indexName.toLowerCase()
+      );
+
+      return ip !== undefined ? ip.id : undefined;
+    } catch (error) {
+      throw error;
```

Comment:
these try/catches are redundant as just throwing `error` is the same as not having the try/catch

Reply:
Removed in https://github.com/elastic/kibana/pull/66934/commits/ba820793f855ac45c4109ec490bf0cfce1a98fcc